### ISSUE TITLE
502 Bad Gateway

### ISSUE THREAD

**Issue by dnohr** [fastify/fastify#3752]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.27.2

### Node.js version

17.6.0

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

Ubuntu 20.04.4

### Description

Hi,

I have a server with node 16.4 with a few sites powered by Fastify or Express.

I would like to upgrade to the latest node 17.6, but I'm getting 502 (Bad gateway) after upgrading on my Fastify powered site, while the Express ones are still working.

All sites are proxied behind NGINX and managed by PM2. When testing locally with nodemon the fastify site works fine with the latest node and on the specified port.

Any suggestions? I'm not sure how to debug it further on the server?

Thanks.

### Steps to Reproduce

Node 17.6, Ubuntu 20.4, NGINX, PM2

**Comment by mcollina** [comment]
Thanks for reporting! Node.js v16 is LTS, so it's the one recommended for production usage. You might want to stick to that.

Note that the inability to update Node.js in this system is a direct result of your choice of using pm2. You want each of your applications to ship their own version of Node.js and you can achieve that by using containers.

---

Going into the problem in detail.. it's very hard to help diagnose the problem without access to the system. Could you build a repro by using containers?

The fundamental question you would need to answer is why nginx is returning a 502. A good place to start are the application logs and the nginx logs.

Could you try checking if v17.0.0 have the same problem?

**Comment by dnohr** [comment]
Hi Matteo,

First of all, thanks for working on Fastify, it's great!

You're right, everything worked fine on Node v16 and it's recommended for production.

After a bit of playing around I managed to make it work on Node v17.0.0+. I'm not sure why the old line didn't work, but I changed following line and then it was reachable:

`fastify.listen(3000)` to `fastify.listen({ port: 3000, host: '0.0.0.0' })`

For now my case is solved, but maybe there's a bug or something?

Thanks a lot.
